src/Dfe.PlanTech.Application/Submission/Commands/SubmitAnswerCommand.cs

```diff
@@ -0,0 +1,116 @@
+using Dfe.PlanTech.Application.Submission.Queries;
+using Dfe.PlanTech.Domain.Answers.Models;
+using Dfe.PlanTech.Domain.Questionnaire.Models;
+using Dfe.PlanTech.Domain.Questions.Models;
+using Dfe.PlanTech.Domain.Responses.Models;
+
+namespace Dfe.PlanTech.Application.Submission.Commands
+{
+    public class SubmitAnswerCommand
+    {
+        private readonly GetSubmitAnswerQueries _getSubmitAnswerQueries;
+        private readonly RecordSubmitAnswerCommands _recordSubmitAnswerCommands;
+
+        public SubmitAnswerCommand(GetSubmitAnswerQueries getSubmitAnswerQueries, RecordSubmitAnswerCommands recordSubmitAnswerCommands)
+        {
+            _getSubmitAnswerQueries = getSubmitAnswerQueries;
+            _recordSubmitAnswerCommands = recordSubmitAnswerCommands;
+        }
+
+        public async Task<int> SubmitAnswer(SubmitAnswerDto submitAnswerDto, string sectionId, string sectionName)
+        {
+            int userId = await _getSubmitAnswerQueries.GetUserId();
+
+            int submissionId;
+
+            if (submitAnswerDto.SubmissionId == null || submitAnswerDto.SubmissionId == 0)
```

Comment:
Would it be worth moving this if statement (the one that gets/sets `submissionId`) to its own method, and just going `int submissionId = GetSubmissionId()` or whatever?

Reply:
Resolved in: [eec9097](https://github.com/DFE-Digital/plan-technology-for-your-school/pull/151/commits/eec909719de3743352bc97c25946eaff9ac540f1)